web/src/main/java/org/springframework/security/web/header/writers/CompositeHeaderWriter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.web.header.writers;
+
+import org.springframework.security.web.header.HeaderWriter;
+import org.springframework.util.Assert;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Collection;
+/**
+ * Provide composite for list of HeaderWriter.
+ *
+ * @author Ankur Pathak
+ * @since 5.2.0
+ */
+public class CompositeHeaderWriter implements HeaderWriter {
+
+	private final Collection<HeaderWriter> headerWriters;
+
+	/**
+	 * Creates a new instance.
+	 *
+	 * @param headerWriters the {@link HeaderWriter} instances to write out headers to the
+	 *                      {@link HttpServletResponse}.
+	 */
+	public CompositeHeaderWriter(Collection<HeaderWriter> headerWriters) {
+		Assert.notNull(headerWriters, "headerWriters cannot be null");
+		this.headerWriters = headerWriters;
+	}
+
+	@Override
+	public void writeHeaders(HttpServletRequest request, HttpServletResponse response) {
+		headerWriters.forEach(headerWriter -> headerWriter.writeHeaders(request, response));
```

Comment:
Just a quick reminder here, too, about `this`, so

```suggestion
		this.headerWriters.forEach(headerWriter -> headerWriter.writeHeaders(request, response));
```